### PR TITLE
Remove old pre-migration data integrity checks

### PR DESCRIPTION
**Pull request by uranusjr**
These checks were put into place early in 2.x to add stricter constraints in the database. They are executed before the migrations when the user runs 'airflow upgradedb' to make sure there are no data integrity issues that would fail the migration halfway, which is both difficult to debug and recover from.

The checks always pass if the user is on a more recent Airflow 2 version, on which the constraints are now enforced by the database. Since we have removed all migration files prior to 2.10, the user must upgrade to at least 2.10 first (which has the checks) before upgrading to 3.0 anyway, so we can safely remove the checks going forward.

(Additional context: these checks need to be modified if we are to rename `execution_date` on DagRun, which I’m doing in #41818. Instead of needing to work on the change, I think we should remove those checks since they don’t ever need to be run anyway.)

### CODE REVIEW

**Comment by uranusjr** [comment]
I decided to keep the mechanism around integrity checks since we’re very likely to need it again at some point in the future. Only the actual checks were removed (so the mechanism simply runs an empty loop and exits).

**Review by jscheffl — APPROVED** [review]

File: airflow/utils/db.py [hunk 1]
```diff
@@ -1079,442 +1020,10 @@ def reflect_tables(tables: list[MappedClassProtocol | str] | None, session):
...
 def _check_migration_errors(session: Session = NEW_SESSION) -> Iterable[str]:
     """:session: session of the sqlalchemy."""
-    check_functions: tuple[Callable[..., Iterable[str]], ...] = (
-        check_conn_id_duplicates,
-        check_conn_type_null,
-        check_run_id_null,
-        check_bad_references,
-        check_username_duplicates,
-    )
+    check_functions: Iterable[Callable[..., Iterable[str]]] = ()
```
Comment (sunank200) [inline airflow/utils/db.py:1026]: Should we remove `_check_migration_errors` completely as it is empty right now and we can add in future when needed?

**Review by sunank200 — APPROVED** [review]
Overall LGTM.

File: airflow/utils/db.py (hunk 1 above)
Comment (uranusjr) [reply airflow/utils/db.py:1026]: I was thinking it would be needed in the future anyway and it’s easier to just keep this here.

**Pull request merged**